Work in progress API Migrate

VulpoTheDev opened the pull request.

kuroji-fusky committed:
ugh the navbar is so MESSY

kuroji-fusky committed:
refact navbar for readability

kuroji-fusky committed:
temp disable fetchdata because of errors

VulpoTheDev committed:
Work in progress API Migrate

VulpoTheDev committed:
Revert navbar api for now

VulpoTheDev committed:
Temp redirect to / after successful login

VulpoTheDev committed:
Image Render

VulpoTheDev committed:
Updated UserType

VulpoTheDev committed:
Display User Data in Navbar

VulpoTheDev committed:
Redirect to / if logged in

kuroji-fusky commented on apps/website/src/app/(main)/login/page.tsx:
```diff
@@ -1,10 +1,14 @@
-export default function SignIn() {
+export default async function SignIn() {
+  const user = fetchUserData()
+  if (user) redirect("/")
```
This is okay but you can use `middleware.ts` and transfer this logic there instead, it might be jarring to go the `/login` route first, then see the login page for a split second, then redirect it to `/`

Reference: <https://nextjs.org/docs/app/building-your-application/routing/middleware>

VulpoTheDev replied:
I gotchu from my observance it fetches the data and then display the page data. I'll work on getting middlewares in after MVP

kuroji-fusky committed:
implemented basic artwork page

kuroji-fusky committed:
Merge branch 'art-page' into frontend-api

kuroji-fusky committed:
impl comment component with react-quill and stuff

kuroji-fusky committed:
refactor dashboard components

kuroji-fusky committed:
fix code style

kuroji-fusky committed:
removed all context providers

closes #44

VulpoTheDev committed:
Banners, Update Default stuff fix pages not loading

VulpoTheDev committed:
Ignore unauth for register

VulpoTheDev committed:
Add email verification pages

VulpoTheDev committed:
Remove console.log statements and add link to user profile in NavbarUserActions

VulpoTheDev committed:
Update DashboardLayout and related components to fetch user data and display it in the navbar and sidebar

kuroji-fusky committed:
refactor env and backend logic with proper typing

kuroji-fusky committed:
linting

VulpoTheDev committed:
Connected CharacterView component with the api and add fetchUserCharacters function

VulpoTheDev committed:
Fix API error handling and remove console.log

VulpoTheDev committed:
WIP Character Page

kuroji-fusky committed:
more refactoring, add middleware

kuroji-fusky committed:
Merge 4 commits

- moving some stuff around
- move and lazy-load dashboard components
- type stuff, init react query provider
- remove axios package

kuroji-fusky committed:
Merge branch 'art-page' into frontend-api

kuroji-fusky committed:
fix stuff after conflict

kuroji-fusky committed:
fixed problems with tanstack query

kuroji-fusky committed:
isolate login stuff, minor dashboard UI changes

kuroji-fusky committed:
utilize tailwind-merge and refactor to use the cn func

kuroji-fusky committed:
add <div> on root layout to compensate dom shifts

kuroji-fusky committed:
rollback api.ts changes, minor ui changes

kuroji-fusky committed:
enable csp rules only on production

kuroji-fusky commented:
fixes #48

kuroji-fusky committed:
overhauled login page

kuroji-fusky committed:
add CharacterView, still wip

kuroji-fusky committed:
refactor button props

kuroji-fusky committed:
whitelist `localhost` from csp header

VulpoTheDev committed:
Heavy WIP Character Creation Page

VulpoTheDev committed:
Refactored DropZone

kuroji-fusky committed:
cleanup redundant React imports

kuroji-fusky committed:
update eslint rules to forbid `<img>` element

kuroji-fusky committed:
refac'd register form and auth pages

kuroji-fusky committed:
export custom hooks from index

VulpoTheDev committed:
Improved image upload functionality to DropZone component

VulpoTheDev committed:
WIP Character Creation Page

kuroji-fusky committed:
lots of minor changes

- add prefetch and preconnect to localhost backends
- lazy load framer motion anims from login page
- update root navbar to remove route checking
- update umami analytics url

kuroji-fusky committed:
remove unused mock imgs

kuroji-fusky committed:
redirected userName.handle route correctly to its @ counterpart

- add gallery page
- block settings and dashboard route from being indexed from search engine
- add gallery page (wip)

VulpoTheDev committed:
Merge branch 'frontend-api' of github.com:MyArtverse-Project/MyArtverse into character-creation-page

kuroji-fusky committed:
add character edit page and change sidebar state

- add gallery page
- block settings and dashboard route from being indexed from search engine
- redirected userName.handle route correctly to its @ counterpart
- dashboard: add gallery page (wip)
- add skip to content for accessibility

VulpoTheDev committed:
Refactor RichTextField component and remove unused files

VulpoTheDev committed:
Merge branch 'frontend-api' of github.com:MyArtverse-Project/MyArtverse into character-creation-page

VulpoTheDev committed:
Merge branch 'character-creation-page' of github.com:MyArtverse-Project/MyArtverse into character-creation-page

kuroji-fusky committed:
add CheckLocalSettings to check system themes and a11y

VulpoTheDev committed:
...

VulpoTheDev committed:
add character edit page and change sidebar state (#51)

VulpoTheDev committed:
Merge branch 'frontend-api' of github.com:MyArtverse-Project/MyArtverse into character-creation-page

kuroji-fusky committed:
simplify knip config

add `knip` and eslint import rules for maintainability

kuroji-fusky committed:
made dropzone clickable and unified form imports

- minor changes to middleware and sum eslint rules
- load sidebar dashboard client side

VulpoTheDev committed:
Fixes, typos

kuroji-fusky committed:
typo: Varient(s) -> Variant(s); consolidating imports

VulpoTheDev committed:
Added functionaility to modal

VulpoTheDev committed:
WIP Changes before submitting character data to api

VulpoTheDev committed:
Simplified Character Creation to Modal, WIP Character Creation to API

VulpoTheDev committed:
Connected API and Edit Character Page WIP

VulpoTheDev committed:
Refactor character edit functionality and fix typo in CharacterTable

VulpoTheDev committed:
Refactor character edit functionality and fix typo in CharacterTable

VulpoTheDev committed:
WIP Character Creation Page (#50)

VulpoTheDev committed:
Redirect user from Home -> Browse page on Log In

VulpoTheDev committed:
Improved Modal

VulpoTheDev committed:
Updated Comments to mostly match design -- WIP Profile Comments

VulpoTheDev committed:
Updated Contributors & Copyright

VulpoTheDev committed:
Refactor code to remove unnecessary console.log statements and improve code readability

VulpoTheDev committed:
Updated Pinned character component and moved Art page under character

VulpoTheDev committed:
Moved profile files around for character pages

VulpoTheDev committed:
Character Page

VulpoTheDev committed:
Browse Page, Featured & New Character API interact

VulpoTheDev committed:
Refactor code to remove unnecessary console.log statements, improve code readability, and update profile files for character pages

VulpoTheDev committed:
Fav function, changed button func

VulpoTheDev committed:
Added pronouns, character species and genders along with visual updates

VulpoTheDev committed:
WIP Gallery Page and Upload Modal

VulpoTheDev committed:
Fixed "That one error"

VulpoTheDev committed:
Use Artwork component instead of div

VulpoTheDev committed:
Display Artwork Details

VulpoTheDev committed:
Heavy WIP Deletion Character

VulpoTheDev committed:
Delete Modal

VulpoTheDev committed:
Art page updates, WIP Edit Page

VulpoTheDev committed:
Commenting on Profiles

VulpoTheDev committed:
Character and Artwork Comments
